Rendera/include/cube.hpp

```diff
@@ -0,0 +1,181 @@
+{
+    // if dot of a-ray.origin and ray.direction is smaller than dot of b-ray.origin and ray.direction
+    if (dot((a - r.get_origin()),r.get_direction()) < dot((b- r.get_origin()),r.get_direction()))
+    {
+        return a;
+    }
+    else
+    {
+        return b;
+    }
+}
+
+class Cube
+{
+private:
+    vec3 m_position;
+    vec3 m_a, m_b, m_c;
+    Material m_material;
+
+public:
+    Cube(vec3 position, vec3 a, vec3 b, vec3 c, Material mat) : m_position{position}, m_a{a}, m_b{b}, m_c{c}, m_material{mat} {};
+
+    ray normal(const vec3 &point)
+    {
+        // p_pos : vector from m_position to point
+        // np_pos : vector from diagonally opposite corner to point
+        vec3 p_pos=point - m_position;
+        vec3 np_pos=point - m_a - m_b - m_c - m_position;
+        vec3 normal;
+
+        // every if statement checks if the point is on which of the 6 faces of the cube
+        if (dot(p_pos, cross(m_a, m_b)) < 1e5)
+        {
+            normal = cross(m_a, m_b);
+            if (dot(m_c, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (abs(dot(p_pos, cross(m_a, m_c))) < 1e-5)
+        {
+            normal = cross(m_a, m_c);
+            if (dot(m_b, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(p_pos, cross(m_b, m_c)) < 1e5)
+        {
+            normal = cross(m_b, m_c);
+            if (dot(m_a, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(np_pos, cross(m_a, m_b)) < 1e5)
+        {
+            normal = cross(m_a, m_b);
+            if (dot(-m_c, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(np_pos, cross(m_a, m_c)) < 1e5)
+        {
+            normal = cross(m_a, m_c);
+            if (dot(-m_b, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(np_pos, cross(m_b, m_c)) < 1e5)
+        {
+            normal = cross(m_b, m_c);
+            if (dot(-m_a, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else
+        {
+            normal = vec3(0, 0, 0);
+        }
+        return ray(point, normal);
+    }
+
+    // returns the intersection of the ray with the plane passing through the point p and perpendicular to the normal n
+    vec3 plane_intersect(const vec3 &n, const vec3 &p, const ray &r)
+    {
+        float num = dot(n, p - r.get_origin());
+        if (abs(dot(r.get_direction(), n)) > 1e-5)
+        {
+            num = num / dot(r.get_direction(), n);
+        }
+        else                                            // if ray is parallel to the plane
+        {
+            if(num > 0){
+                num = std::numeric_limits<float>::max();
+            }
+            else{
+                num = std::numeric_limits<float>::lowest();
```

Comment:
Any reason to use `lowest()` here instead of `min()` here?

Reply:
min returns value close to zero value while lowest return lowest negative value when using float
 [reference](https://en.cppreference.com/w/cpp/types/numeric_limits) - last example